File: android/app/src/main/java/my/ssdid/wallet/domain/oid4vp/CredentialQuery.kt
package my.ssdid.wallet.domain.oid4vp

data class CredentialQuery(
    val descriptors: List<CredentialQueryDescriptor>
)

data class CredentialQueryDescriptor(
    val id: String,
    val format: String,
    val vctFilter: String?,
    val requiredClaims: List<String>,
    val optionalClaims: List<String>
)

data class MatchResult(
    val descriptorId: String,
    val credentialId: String,
    val credentialType: String,
    val availableClaims: Map<String, ClaimInfo>,
    val source: CredentialSource
)

data class ClaimInfo(
    val name: String,
    val required: Boolean,
    val available: Boolean
)

enum class CredentialSource {
    SD_JWT_VC,
    IDENTITY
}

File: sdk/ios/SsdidCore/Sources/SsdidCore/Domain/OpenId4Vp/CredentialQuery.swift
import Foundation

public struct CredentialQuery {
    public let descriptors: [CredentialQueryDescriptor]
}

public struct CredentialQueryDescriptor {
    public let id: String
    public let format: String
    public let vctFilter: String?
    public let requiredClaims: [String]
    public let optionalClaims: [String]
}

public struct MatchResult {
    public let descriptorId: String
    public let credentialId: String
    public let credentialType: String
    public let availableClaims: [String: ClaimInfo]
    public let source: CredentialSource
}

public struct ClaimInfo {
    public let name: String
    public let required: Bool
    public let available: Bool
}

public enum CredentialSource {
    case sdJwtVc
    case identity
}
